WordPress/Classes/Models/Blog+Lookup.swift

```diff
@@ -58,4 +69,83 @@ public extension Blog {
 
         return service.findBlog(withXmlrpc: xmlrpc, andUsername: username)
     }
+
+    @objc(countInContext:)
+    static func count(in context: NSManagedObjectContext) -> Int {
+        BlogQuery().count(in: context)
+    }
+
+    @objc(wpComBlogCountInContext:)
+    static func wpComBlogCount(in context: NSManagedObjectContext) -> Int {
+        BlogQuery().hostedByWPCom(true).count(in: context)
+    }
+
+    @objc(selfHostedInContext:)
+    static func selfHosted(in context: NSManagedObjectContext) -> [Blog] {
+        (try? BlogQuery().hostedByWPCom(false).blogs(in: context)) ?? []
+    }
+}
+
+struct BlogQuery {
```

Comment:
Did you consider having this in a dedicated file?

Reply:
I'm pretty relaxed on whether to use a dedicated file or a relevant existing files(kinda reasonable to put a query type in a file with "Lookup" in its name?) for types. But I'll move this type out to its own file, to follow our convention.

Reply:
> kinda reasonable to put a query type in a file with "Lookup" in its name?

Totally reasonable, yeah.

Thank you for considering my suggestion.